### ISSUE TITLE
Template/Command list screen shows wrong VERSION

### ISSUE THREAD

**Issue by yoshwata** [screwdriver-cd/screwdriver#1723]
**What happened**:
The `VERSION` number on template's list screen is something wrong.

The `VERSION` indicates `0.0.9` on screen.
![YUS7i5d5e3560456479e30b8c48da](https://user-images.githubusercontent.com/32473622/63491144-6ad4bc80-c4f1-11e9-8b54-afd269c86fa1.png)

However, there is no special tags like `stable` or `latest` on `0.0.9`, and it also not to be newest one (`0.0.14`).
![Screen Shot 2019-08-22 at 15 27 28](https://user-images.githubusercontent.com/32473622/63491163-77591500-c4f1-11e9-9cd4-5743e0039681.jpg)

**What you expected to happen**:
`latest`, `stable` tagged version or simply newest one should appear on list screen.
(I prefer to show `stable` version.)

**How to reproduce it**:
1. Register a template and update 10 time to make the version `0.0.10`.
2. Set tags `stable` and `latest` to `0.0.10`
3. Then, the `VERSION` of list screen is still `0.0.9`.

This will occur on command's list screen too.

**Comment by yoshwata** [comment]
The cause of problem is, like I wrote at https://github.com/screwdriver-cd/screwdriver/pull/1778, the columns is aggregated with `MAX`. 

When we hava a template like below.

| namespace | name | description | version
----|---- |---- |---- |
| foo | bar | 2 | 0.0.10
| foo | bar | 1 | 0.0.9
| foo | bar | 9 | 0.0.8

List screen shows like this about `foo/bar` template (version might be already hidden)

| namespace | name | description | version
----|---- |---- |---- |
| foo | bar | 9 | 0.0.9

The expected behavior is like this.

| namespace | name | description | version
----|---- |---- |---- |
| foo | bar | 2 | 0.0.10

**Comment by yoshwata** [comment]
I propose two solution. I'd like to get your opinion.

#### 1. Paginating

Restricting number of DB queries by paginating. New search queries are emitted by scrolling list screen, so DB performance is not so bad.

#### 2. Implement DB query to get the list of newest templates.

This is a sample for mysql to list newest version of templates. 

```sql
select namespace, name, (select version from templates where namespace=t.namespace and name=t.name order by INET_ATON(SUBSTRING_INDEX(CONCAT(version, '.0'),'.',4)) desc limit 1) version from templates t group by namespace, name;
```

However it does not work for number grater than 255 because of INET_ATON. We have to rethink DB query to proceed this plan.

**Comment by jithin1987** [comment]
Option 2 appears to be simplest and easiest to implement, should cover 90% of cases. 

For postgres this is an option https://dba.stackexchange.com/questions/74283/how-to-order-by-typical-software-release-versions-like-x-y-z

But to solve this generically option 4 seems to be right approach.

**Comment by wahapo** [comment]
Option 2: This is an interesting implementation. But ORACLE does not have INET_ATON. To implement this option, it seems necessary to switch the process of [sequelize](https://github.com/screwdriver-cd/datastore-sequelize/blob/master/index.js) depending on the type of RDB. I am worried that this would make the code very dirty.

Option 3: I feel that it will be a simple implementation compared to option 2. I expect it to be a modification that just uses a subquery. And there are almost no cases in Yahoo! JAPAN where a template with a lower version number is published as the latest version. It's a very special case, and if it exists, it will be better than current one.

Option 4: I expect that the subquery part modified with option 3 will be extended to use three columns. So, I feel that it is a good correction volume to work on after option 3.(However, it is quite a volume ...)

@jithin1987 I want to start with Option 3 to advance development to Option 4. If you have any concerns about Option 3, would you provide feedback?

**Comment by jithin1987** [comment]
@wahapo I am good with option 3. Actually in my previous comment I meant to say option 3 only 🤦‍♂

**Comment by wahapo** [comment]
@jithine OK, then first of all I will implement with option 3. Thank you for your feedback😃

**Comment by wahapo** [comment]
By research, I discovered a pattern where createTime is NULL. In this case I need to look for other plausible sort key. It seems that there is nothing appropriate other than id.
```
> select id, name, namespace, createTime from templates;
+----+------------------+-------------+------------+
| id | name             | namespace   | createTime |
+----+------------------+-------------+------------+
|  1 | foo              | bar         | NULL       |
|  2 | foo              | bar         | NULL       |
+----+------------------+-------------+------------+
```

By using the following query, I can achieve what I want to achieve fairly accurately.
```
select id, name, namespace, createTime from templates as m where not exists(select 1 from templates as s where m.name = s.name and m.namespace = s.namespace and m.id < s.id);

+----+------------------+-------------+------------+
| id | name             | namespace   | createTime |
+----+------------------+-------------+------------+
|  2 | foo              | bar         | NULL       |
+----+------------------+-------------+------------+
```

**Comment by wahapo** [comment]
Sequelize is unlikely to handle subqueries well. I thought another SQL, so I will try it.
```
select
    templates.id,
    templates.name,
    templates.namespace,
    templates.version,
    templates.createTime
  from templates
  left join templates as t on (
    templates.name = t.name
    and templates.namespace = t.namespace
    and templates.id < t.id
  )
  where t.id is null;
```

**Comment by wahapo** [comment]
It seems that association setting is necessary to use `LEFT JOIN` with sequelize.
https://sequelize.org/master/manual/associations.html

When setting association between the same table names, I gave another table name `t` with `AS` clause, however when I accessed it twice in the browser, it shows alias `t` already exists, so it didn't work. In the case of joining tables with the same name, I didn't find a way to solve this well.

I found an example using subquery and implemented it with reference to this.
https://stackoverflow.com/questions/36164694/sequelize-subquery-in-where-clause
